diff --git a/src/Controller/DiscourseSsoController.php b/src/Controller/DiscourseSsoController.php
index ab5bec4..bd69562 100644
--- a/src/Controller/DiscourseSsoController.php
+++ b/src/Controller/DiscourseSsoController.php
@@ -8,6 +8,7 @@ use Drupal\Core\Access\AccessResult;
 use Drupal\Core\Config\ImmutableConfig;
 use Drupal\Core\Controller\ControllerBase;
 use Drupal\Core\Routing\TrustedRedirectResponse;
+use Drupal\Core\Url;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 
@@ -150,8 +151,13 @@ class DiscourseSsoController extends ControllerBase {
     $_SESSION['discourse_sso_sig'] = $this->sig;
     $_SESSION['discourse_sso_payload'] = $this->payload;
 
+    $url = Url::fromRoute('discourse_sso.sso', [], [
+      'sso' => $this->payload,
+      'sig' => $this->sig,
+    ]);
+
     $options = [
-      'query' => ['destination' => 'discourse_sso?sso=' . $this->payload . '&sig=' . $this->sig],
+      'query' => ['destination' => $url->toString()],
       'absolute' => TRUE,
     ];
     return $this->redirect('user.login', [], $options);
